### DISCUSSION TITLE
[solved] - Trying to install operator

### DISCUSSION THREAD

**Discussion by smolinari** [alexandrevilain/temporal-operator#716] [General]
Hi,

I'm trying to follow the directions [from the docs site](https://temporal-operator.pages.dev/getting-started/) to install the operator. I have already gotten pgres and elasticsearch servers running. For the installation, I'mt using [the es and pg example](https://github.com/alexandrevilain/temporal-operator/tree/main/examples/cluster-postgres-es).

First problem I ran into was 

> The TemporalCluster "dev" is invalid: spec.persistence.advancedVisibilityStore.elasticsearch.indices: Invalid value: "string": spec.persistence.advancedVisibilityStore.elasticsearch.indices in body must be of type object: "string"

 
 So, I went looking into the chart code and saw that it looks for `indices.Visibility` so I added that to the values of the `TemporalCluster` values. 
 
 When I ran kubectl apply, I got the temporalcluster is created, but no pods are starting. 
 
 What did I do wrong?
 
 Scott

**Comment by smolinari** [comment]
Hmm.. Ok. I inadvertently forgot to run the most important apply for the operator itself. :man_facepalming: 

I did that, now I have this error:

> The TemporalCluster "dev" is invalid: spec.version: Forbidden: Unsupported temporal version (supported: >=1.14.0 <1.23.0)

I'll try lowering the version, but it seems like an error in the chart too?

Scott

**Reply by alexandrevilain** [reply]
Hi @smolinari !

Maybe you're trying to install temporal v1.23.0 ? It will be released during this week in the operator v0.18.0.

**Comment by smolinari** [comment]
Ok. Changed the version to 1.22.7 and things started to run. However, I'm now stuck with the Job pod `dev-update-advanced-visibility-schema-v-1-22-7-65vmg` and it is saying this in the logs:

> Can't do upgrade from v0 schema, version needing advanced visibility schema v1 are not supported by the operator

So, now I'm genuinely stuck. Any tips for troubleshooting or possible solutions would be greatly appreciated!

Scott

**Reply by alexandrevilain** [reply]
Could you please ensure that your `ES`cluster is empty ? Looks like it has data ?

**Reply by smolinari** [reply]
Hi Alexandre,

Thanks for replying. 

Does the whole cluster need to be empty? Or just the index? The index was newly created for the Temporal operator installation. I also have another index for the "original" Temporal k8s helm install, which wasn't working for me.   

Scott

**Reply by smolinari** [reply]
Ok. Nevermind. I deleted the index and created it again and now the installation seems to be continuing. 

Scott

**Reply by smolinari** [reply]
This is what has started.

![image](https://github.com/alexandrevilain/temporal-operator/assets/3314957/11e3a684-d822-44f5-92a5-6d8bb581cb84)

I'll continue to work with this. Thanks!

Scott

**Reply by alexandrevilain** [reply]
Cool! 
Feel free to create issues if something is going wrong. 
Please note that 1.23+ support is comming in few days.

**Reply by smolinari** [reply]
Awesome!!! 

Scott
